### ISSUE TITLE
[REVIEW]: 3D Line Radiative Transfer & Synthetic Observations with Magritte

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3905]
**Submitting author:** @FredDeCeuster (<a href="http://orcid.org/0000-0001-5887-8498">Frederik De Ceuster</a>)
**Repository:** https://github.com/Magritte-code/Magritte
**Branch with paper.md** (empty if default branch): joss_paper_1.0
**Version:** 0.0.1
**Editor:** @xuanxu
**Reviewers:** @gonsie, @CFGrote
**Archive:** 10.5281/zenodo.6320469

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/010e408b39422f6691f876925e37d256/status.svg)](https://joss.theoj.org/papers/010e408b39422f6691f876925e37d256)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/010e408b39422f6691f876925e37d256"><img src="https://joss.theoj.org/papers/010e408b39422f6691f876925e37d256/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/010e408b39422f6691f876925e37d256/status.svg)](https://joss.theoj.org/papers/010e408b39422f6691f876925e37d256)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/Magritte-code/Magritte.git" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@gonsie & @CFGrote, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @xuanxu know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @gonsie
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/Magritte-code/Magritte.git">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Magritte-code/Magritte.git">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@FredDeCeuster) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @CFGrote
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/Magritte-code/Magritte.git">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Magritte-code/Magritte.git">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@FredDeCeuster) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @gonsie, @CFGrote it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
:wave: @CFGrote, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @gonsie, please update us on how your review is going (this is an automated reminder).

**Comment by gonsie** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3905 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by xuanxu** [comment]
@whedon generate pdf from branch joss_paper_1.0

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_paper_1.0. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03905/joss.03905/10.21105.joss.03905.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03905/joss.03905/10.21105.joss.03905.pdf) :page_facing_up: :point_left:

**Comment by gonsie** [comment]
@FredDeCeuster can you clarify the author list on the paper?

In the readme Atulit Srivastava is listed as a developer, but I can't find any contributions from them in GitHub. 

In the readme Ward Homan and Jan Bolte are listed as technical advisors, but they are not included in the author list.

**Comment by FredDeCeuster** [comment]
@gonsie

Atulit Srivastava developed parts for a future version (currently not public), but also already contributed ideas for this version.

Ward Homan and Jan Bolte mostly contributed to the science done with Magritte (see papers in documentation), but I would be happy to include them as co-authors on this paper.

**Comment by CFGrote** [comment]
Review started. Installation went ok with minor hickups (will post separate issue later). Running the first example gave me errors, see https://github.com/Magritte-code/Magritte/issues/54. I'd like to see this fixed before I continue.

**Comment by gonsie** [comment]
Thanks @FredDeCeuster, I think the author list you have now is fine, I just wanted to clarify.

**Comment by CFGrote** [comment]
Review comments on Documentation and Examples in https://github.com/Magritte-code/Magritte/issues/56

**Comment by CFGrote** [comment]
@whedon generat pdf from branch joss_paper_1.0

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by CFGrote** [comment]
@whedon generate pdf from branch joss_paper_1.0

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_paper_1.0. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03905/joss.03905/10.21105.joss.03905.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03905/joss.03905/10.21105.joss.03905.pdf) :page_facing_up: :point_left:

**Comment by CFGrote** [comment]
Software paper review in https://github.com/Magritte-code/Magritte/issues/57

**Comment by gonsie** [comment]
I find the documentation of the code base somewhat lacking. There is almost no API documentation. While the examples do a decent job of walking someone through creating a model, the text itself is limited and doesn't explain much. A note in the documentation points to the code base... but there are very few comments in the code.

**Comment by CFGrote** [comment]
@whedon generate pdf from branch joss_paper_1.0

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_paper_1.0. Reticulating splines etc...
```

**Comment by CFGrote** [comment]
@FredDeCeuster , can you address the pdf compile error?

> PDF failed to compile for issue #3905 with the following error:
> 
> ```
>  Cloning into '0f2c2d2aa683fed30ea68102'...
> ```

@FredDeCeuster, can you address this?

**Comment by xuanxu** [comment]
@whedon generate pdf from branch joss_paper_1.0

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_paper_1.0. Reticulating splines etc...
```

**Comment by xuanxu** [comment]
Looks like Whedon successfully compiled the paper in the [preview site](http://whedon.theoj.org/), let's try again

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03905/joss.03905/10.21105.joss.03905.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03905/joss.03905/10.21105.joss.03905.pdf) :page_facing_up: :point_left:

**Comment by CFGrote** [comment]
@FredDeCeuster, just to make sure I'm not missing anything, are you still working on the API documentation or do you see it as ready for a (re-)review?

**Comment by FredDeCeuster** [comment]
Sorry for the late reply.
We've updated the documentation now.
Please let us know if there is anything else we can do.
Many thanks for your feedback!

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_paper_1.0. Reticulating splines etc...
```

**Comment by xuanxu** [comment]
@gonsie @CFGrote please update us on the current status of your reviews

**Comment by CFGrote** [comment]
For me this looks good to go! My issues in https://github.com/Magritte-code/Magritte are resolved and I ticked all boxes above.

**Comment by xuanxu** [comment]
great! thanks @CFGrote

**Comment by xuanxu** [comment]
:wave: @gonsie please update us on the current status of your review

**Comment by gonsie** [comment]
Hi all (@xuanxu @FredDeCeuster), I'm sorry for my delayed response. 

I've updated my checkmarks. Due to the complexity of building this, I am unable to verify the functionality (I do not have easy access to the gcc compiler with a proper python/conda environment). While the paper does not reference performance numbers, they are present in the documentation and (again) I am unable to confirm them.

I've filed issue Magritte-code/Magritte#61 about including details about the analytical benchmark in the docs. I would like this documented since the automated tests run only a few benchmarks. While the `tests/run_all_tests.sh` script refers to 'unit tests' and 'integration tests' those do not exist.

There are a lot of details in the documentation that could be added to the paper to flush it out. Currently the paper consists of only a summary, statement of need, and future work. While more details of the current software capabilities would be nice, it  is not a requirement for a Joss paper. 

Once issue # 61 is complete, I will accept this paper. 

Thank you.

**Comment by gonsie** [comment]
I am satisfied with my review and accept this paper.

**Comment by xuanxu** [comment]
@gonsie thanks!

**Comment by xuanxu** [comment]
@editorialbot generate pdf

**Comment by xuanxu** [comment]
@editorialbot check references

**Comment by xuanxu** [comment]
@editorialbot set joss_paper_1.0 as branch

**Comment by xuanxu** [comment]
@editorialbot check references

**Comment by xuanxu** [comment]
@editorialbot generate pdf

**Comment by xuanxu** [comment]
@editorialbot check references

**Comment by xuanxu** [comment]
@FredDeCeuster the editorial bot suggests [10.1146/annurev-astro-090120-033712](https://doi.org/10.1146/annurev-astro-090120-033712) as the DOI for `Decin, L. (2020)` reference. Can you check if it is the right one and if so update the paper with it?

**Comment by FredDeCeuster** [comment]
I've checked and added the doi for Decin (2020).
https://github.com/Magritte-code/Magritte/commit/30b5d08536350c997bb30526b094c2612442c277

**Comment by xuanxu** [comment]
@editorialbot generate pdf

**Comment by xuanxu** [comment]
@editorialbot check references

**Comment by xuanxu** [comment]
OK @FredDeCeuster, everything looks good, here are the next steps:

- Please release a new tagged version from the current stable branch so it includes all the changes made during the review process
- Then archive that latest release in Zenodo or a similar service
- Check the Zenodo deposit has the correct metadata: title and authors names should match the paper; you may also add authors' ORCIDs.

Once you do that please report here the version number and archive DOI

**Comment by kyleniemeyer** [comment]
@editorialbot accept

**Comment by kyleniemeyer** [comment]
Congratulations @FredDeCeuster on your article's publication in JOSS!5887-8498))

Many thanks to @gonsie and @CFGrote for reviewing this, and @xuanxu for editing.
